nixos/modules/hardware/cpu/intel-opencl.nix

```diff
@@ -0,0 +1,31 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+{
+  options.hardware.intel.opencl.runtime = lib.mkOption {
+    type = lib.types.nullOr (
+      lib.types.enum [
+        "intel-compute-runtime"
+        "intel-ocl"
+        "beignet"
+      ]
+    );
+    default = null;
+    description = ''
+      Select the Intel OpenCL runtime to use. Choose your runtime based on your Intel CPU/GPU generation.
```

Comment:
I consider this option documentation, also it's the same amount of code to add stuff to extraPackages, this one makes it easier to discover what needs to be done to get opencl running on a given Intel CPU.